src/librustc/mir/interpret/allocation.rs

```diff
@@ -15,16 +15,17 @@ use std::borrow::Cow;
 
 #[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable)]
 pub struct Allocation<Tag=(),Extra=()> {
-    /// The actual bytes of the allocation.
-    /// Note that the bytes of a pointer represent the offset of the pointer
-    pub bytes: Vec<u8>,
+    /// The byte allocation and definedness.
+    /// Bytes of fully undefined allocations are allocated lazily on potentially mutable changes.
+    pub alloc: Option<AllocationBytes>,
     /// Maps from byte addresses to extra data for each pointer.
     /// Only the first byte of a pointer is inserted into the map; i.e.,
     /// every entry in this map applies to `pointer_size` consecutive bytes starting
     /// at the given offset.
     pub relocations: Relocations<Tag>,
```

Comment:
Relocations should also move into `alloc` then, no point in keeping them when the allocation is fully uninitialized.

Reply:
Good to have this confirmed, I was about to ask as parts in 'codegen' don't make much sense in that state.